internal/replacements/replacements.go

```diff
@@ -186,13 +187,17 @@ func processGroupNeedsRemovalForPod(cluster *fdbv1beta2.FoundationDBCluster, pod
 		return true, nil
 	}
 
+	specHash, err := internal.GetPodSpecHash(cluster, processGroupStatus, nil)
+	if err != nil {
+		return false, err
+	}
+	spec, err := internal.GetPodSpec(cluster, processGroupStatus)
+	if err != nil {
+		return false, err
+	}
+	// TODO the below logic could be consolidated further since the majority of them would be skipped if pod Spec has not changed.
```

Comment:
Can you create an issue for that?

Reply:
yeah, I could also do it in this PR if you prefer but it feels like there are too many changes going in already

Reply:
Creating an issue is good enough :) We can work on that in another PR.